src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -85,28 +85,27 @@ private[redshift] class RedshiftWriter(
     val randomSuffix = Math.abs(Random.nextInt()).toString
     val tempTable = s"${table}_staging_$randomSuffix"
     val backupTable = s"${table}_backup_$randomSuffix"
+    log.info("Loading new Redshift data to: " + tempTable)
+    log.info("Existing data will be backed up in: " + backupTable)
 
     try {
-      log.info("Loading new Redshift data to: " + tempTable)
-      log.info("Existing data will be backed up in: " + backupTable)
-
       action(tempTable)
+
       if (jdbcWrapper.tableExists(conn, table)) {
-        conn.prepareStatement(s"ALTER TABLE $table RENAME TO $backupTable").execute()
+        conn.prepareStatement(
+          s"""
+             | BEGIN;
```

Comment:
@jaley, was there any reason that you chose to not use a transaction to perform these renames atomically?